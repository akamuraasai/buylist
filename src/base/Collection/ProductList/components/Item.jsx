import React from 'react';
import PropTypes from 'prop-types';

import moneyFormatter from '../../../../helpers/moneyFormatter';

const toPrice = moneyFormatter;

const Item = ({ name, price, units, unitType, completed }) => (
  <div className={`ProductList__item ${completed ? 'is-complete' : ''}`}>
    <div className="ProductList__item__image">
      <img src="http://via.placeholder.com/60x60" alt="product" />
    </div>
    <div className="ProductList__item__text">
      <p className="ProductList__item__text__title">{name}</p>
      <p><b>R$ {toPrice(price)} </b> x {units}{unitType}</p>
    </div>
    <div className="ProductList__item__price">
      <p>R$ {toPrice(price * units)}</p>
    </div>
  </div>
);

Item.defaultProps = {
  name: '',
  price: 0,
  units: 0,
  unitType: 'un',
  completed: false,
};

Item.propTypes = {
  name: PropTypes.string,
  price: PropTypes.number,
  units: PropTypes.number,
  unitType: PropTypes.string,
  completed: PropTypes.bool,
};

export default Item;
