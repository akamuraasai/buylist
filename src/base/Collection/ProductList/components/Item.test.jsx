import React from 'react';
import { shallow } from 'enzyme';
import Item from './Item';

describe('ProductList > Item', () => {
  it('should render without crashing', () => {
    const item = shallow(<Item />);
    expect(item).toMatchSnapshot();
  });

  it('should render the correct price text', () => {
    const props = {
      price: 12.34,
      units: 10,
    };
    const item = shallow(<Item {...props} />);

    expect(item.find('.ProductList__item__price > p').text()).toEqual('R$ 123,40');
  });

  it('should render a different class if item is completed', () => {
    const props = {
      completed: true,
    };
    const item = shallow(<Item {...props} />);

    expect(item.hasClass('is-complete')).toEqual(true);
  });
});
