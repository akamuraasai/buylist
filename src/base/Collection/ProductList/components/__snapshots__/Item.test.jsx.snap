// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList > Item should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    completed={false}
    name=""
    price={0}
    unitType="un"
    units={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ProductList__item__image"
>
          <img
                    alt="product"
                    src="http://via.placeholder.com/60x60"
          />
</div>,
        <div
          className="ProductList__item__text"
>
          <p
                    className="ProductList__item__text__title"
          />
          <p>
                    <b>
                              R$ 
                              0,00
                               
                    </b>
                     x 
                    0
                    un
          </p>
</div>,
        <div
          className="ProductList__item__price"
>
          <p>
                    R$ 
                    0,00
          </p>
</div>,
      ],
      "className": "ProductList__item ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="product"
            src="http://via.placeholder.com/60x60"
/>,
          "className": "ProductList__item__image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "product",
            "src": "http://via.placeholder.com/60x60",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="ProductList__item__text__title"
/>,
            <p>
              <b>
                            R$ 
                            0,00
                             
              </b>
               x 
              0
              un
</p>,
          ],
          "className": "ProductList__item__text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "ProductList__item__text__title",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  R$ 
                  0,00
                   
</b>,
                " x ",
                0,
                "un",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "R$ ",
                    "0,00",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "R$ ",
                  "0,00",
                  " ",
                ],
                "type": "b",
              },
              " x ",
              0,
              "un",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            R$ 
            0,00
</p>,
          "className": "ProductList__item__price",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "R$ ",
              "0,00",
            ],
          },
          "ref": null,
          "rendered": Array [
            "R$ ",
            "0,00",
          ],
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ProductList__item__image"
>
            <img
                        alt="product"
                        src="http://via.placeholder.com/60x60"
            />
</div>,
          <div
            className="ProductList__item__text"
>
            <p
                        className="ProductList__item__text__title"
            />
            <p>
                        <b>
                                    R$ 
                                    0,00
                                     
                        </b>
                         x 
                        0
                        un
            </p>
</div>,
          <div
            className="ProductList__item__price"
>
            <p>
                        R$ 
                        0,00
            </p>
</div>,
        ],
        "className": "ProductList__item ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="product"
              src="http://via.placeholder.com/60x60"
/>,
            "className": "ProductList__item__image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "product",
              "src": "http://via.placeholder.com/60x60",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="ProductList__item__text__title"
/>,
              <p>
                <b>
                                R$ 
                                0,00
                                 
                </b>
                 x 
                0
                un
</p>,
            ],
            "className": "ProductList__item__text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "ProductList__item__text__title",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    R$ 
                    0,00
                     
</b>,
                  " x ",
                  0,
                  "un",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "R$ ",
                      "0,00",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "R$ ",
                    "0,00",
                    " ",
                  ],
                  "type": "b",
                },
                " x ",
                0,
                "un",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              R$ 
              0,00
</p>,
            "className": "ProductList__item__price",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "R$ ",
                "0,00",
              ],
            },
            "ref": null,
            "rendered": Array [
              "R$ ",
              "0,00",
            ],
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
